Fix error messages

maltelueken opened the pull request:
Fixes the error message for Hayes models that are not implemented to appear correctly (previously did not appear at all).

Fixes the error message when a linear local test type is chosen but a factor variable is in the model.

Removes the footnote about model saturation from the model summary table.

Changes the footnote about model converge to appear as a symbol for the corresponding models in the model summary table.

Adds a disclaimer about causal effects to parameter estimates output.

Kucharssim commented on R/classicProcess.R:
```diff
@@ -1143,7 +1153,7 @@ ClassicProcess <- function(jaspResults, dataset = NULL, options) {
   ))
 
   if (inherits(fittedModel, "try-error")) {
-    return(gettextf("Estimation failed\nMessage:\n%s", attr(fittedModel, "condition")$message))
+    return(gettextf("Estimation failed: %s", gsub("lavaan ERROR:", "", attr(fittedModel, "condition")$message)))
   }
```
this is just a minor point, but you *could* use `jaspBase::isTryError` and `jaspBase::.extractErrorMessage`

Kucharssim reviewed (approved):
The code changes look ok to me, but there are some general issues:

There still some kind of state issue; In this file (change extension from `.jasp` to `.zip`): [process-bug.zip](https://github.com/jasp-stats/jaspProcess/files/13378561/process-bug.zip)

I get `Error in object@Options: no applicable method for '@' applied to an object of class 'NULL'`, right when I created the second model

If I refresh the analysis, I get a different error. Then if I remove the first model, the second model runs fine.

Otherwise, the tests fail quite catastrophically on Ubuntu. I assume you did not have the opportunity to test it there? 

I think we could merge this, but tt would be good to make sure that the module is not broken for Linux users for the release...

maltelueken commented:
Thanks for the review! I fixed the state issue (it was actually a container issue) and the issue with the summary table if one model is not estimated yet (invalid). I also used the `jaspBase` functions where possible.

Turns out that containers for new models were not correctly added to `jaspResults`, so their objects could not be accessed. This is fixed now and I moved all state dependencies to the container (because they are the same for all states in the container).

maltelueken commented:
Regardint the failing tests on Linux: This is happening since JASP moved to R 4.3.1, so I think it might be a seed issue. I will try to run the tests on my WSL to see if they also fail there.

maltelueken commented:
I fixed a bug that caused the large differences between Linux and the other systems for some failed tests. The bug was that covariances where estimated between mediator variables that also had a causal path between them.
The other tests still fail due to differences at the 5th decimal. This will never be visible to the users, so I think it is ok to leave this for now.
